exporter/chronicleexporter/marshal.go

```diff
@@ -482,26 +519,60 @@ func buildForwarderString(cfg Config) string {
 	return fmt.Sprintf(format, cfg.Project, cfg.Location, cfg.CustomerID, cfg.Forwarder)
 }
 
-func (m *protoMarshaler) constructHTTPPayloads(rawLogs map[string][]*api.Log) map[string]*api.ImportLogsRequest {
-	payloads := make(map[string]*api.ImportLogsRequest, len(rawLogs))
+func (m *protoMarshaler) constructHTTPPayloads(rawLogs map[string][]*api.Log) map[string][]*api.ImportLogsRequest {
+	payloads := make(map[string][]*api.ImportLogsRequest, len(rawLogs))
 
 	for logType, entries := range rawLogs {
 		if len(entries) > 0 {
-			payloads[logType] =
-				&api.ImportLogsRequest{
-					// TODO: Add parent and hint
-					// We don't yet have solid guidance on what these should be
-					Parent: "",
-					Hint:   "",
-
-					Source: &api.ImportLogsRequest_InlineSource{
-						InlineSource: &api.ImportLogsRequest_LogsInlineSource{
-							Forwarder: buildForwarderString(m.cfg),
-							Logs:      entries,
-						},
-					},
-				}
+			request := m.buildHTTPRequest(entries)
+
+			payloads[logType] = m.enforceMaximumsHTTPRequest(request)
 		}
 	}
 	return payloads
 }
+
+func (m *protoMarshaler) enforceMaximumsHTTPRequest(request *api.ImportLogsRequest) []*api.ImportLogsRequest {
+	size := proto.Size(request)
+	logs := request.GetInlineSource().Logs
+	if size <= m.cfg.BatchRequestSizeLimitHTTP && len(logs) <= m.cfg.BatchLogCountLimitHTTP {
+		return []*api.ImportLogsRequest{
+			request,
+		}
+	}
+
+	if len(logs) < 2 {
+		m.teleSettings.Logger.Error("Single entry exceeds max request size. Dropping entry", zap.Int("size", size))
+		return []*api.ImportLogsRequest{}
```

Comment:
I'm not sure I'm a fan of this, but I'm assuming this is the current behavior of secops where it will just drop an entry if it's too big?

Reply:
Yeah if one entry were too big for SecOps, it would just reject the request. Not sure what the alternative to this approach would be - this is what we did for Stanza as well sending to Google Cloud https://github.com/observIQ/stanza/blob/main/operator/builtin/output/googlecloud/request.go#L42

Reply:
Cool, if logs are going to be dropped regardless by secops then this is fine with me.